crates/noirc_abi/src/lib.rs

```diff
@@ -208,19 +229,39 @@ impl Abi {
             })
             .collect();
 
-        Ok(witness_map)
-    }
-
-    /// Checks that no extra witness values have been provided.
-    fn check_for_unexpected_inputs(&self, inputs: &InputMap) -> Result<(), AbiError> {
-        let param_names = self.parameter_names();
-        if param_names.len() < inputs.len() {
-            let unexpected_params: Vec<String> =
-                inputs.keys().filter(|param| !param_names.contains(param)).cloned().collect();
-            return Err(AbiError::UnexpectedParams(unexpected_params));
+        // The user can optionally provide a return value to be inserted into the witness map.
```

Comment:
How does the user optionally provide a return value from the toml file?

Reply:
This may be getting some wires crossed due to https://github.com/noir-lang/noir/pull/867. This optional return value is to differentiate between encoding the partial witness for executing the program (no return value passed) or encoding the public inputs to pass to the verifier (potentially with return value passed). It's not optional in the toml file however, it either needs to be there or can't exist.

I can reword this to be a bit clearer though.

Reply:
yeah I had just finished reviewing that so the word "user" was getting my wires crossed